### ISSUE TITLE
image retrieval demo

### ISSUE THREAD

**Issue by YunShen66666** [AmirSh15/FECNet#4]
hi, I'm try to extract the expression information for _expression transfer_. But the model don't works well in croped CelebA-HQ dataset when I try to use a quary image with a expression _A_ to search other images. The ideal result is that images which is  choosed should have the same expression _A_ . However the result is not good. 
For retrieval I just get the target expression tensor though FECNet  ,then choose 100 images randomly and get the output though FECNet. Then I compute the _L2_loss_ between the target image and the choosed images using _torch.nn.functional.mse_loss_ . After get these loss , I sort them and choose the smallest as the nearest expression image. 
I'm not good at image retrieval and don't konw if these method works . Anyway , can you release your retrieval demo code? plz

**Comment by YunShen66666** [comment]
this my retrieval code
`def find(img,path,model):
    target_img_emb = getEmbedding(img,'cpu',model)
    min_emb = []
    for i in range(100):
        choose = random.randint(0,30000)
        img_path = path + "\\" + f"{str(choose).zfill(5)}.jpg"
        img1 = cv2.imread(img_path)
        img1_emb = getEmbedding(img1,'cpu',model)
        loss = F.mse_loss(img1_emb,target_img_emb)
        min_emb.append([loss.item(),choose])
    min_emb.sort()
    return min_emb[:10]
`

**Comment by YunShen66666** [comment]
this my retrieval code
···
def find(img,path,model):
    target_img_emb = getEmbedding(img,'cpu',model)
    min_emb = []
    for i in range(100):
        choose = random.randint(0,30000)
        img_path = path + "\\" + f"{str(choose).zfill(5)}.jpg"
        img1 = cv2.imread(img_path)
        img1_emb = getEmbedding(img1,'cpu',model)
        loss = F.mse_loss(img1_emb,target_img_emb)
        min_emb.append([loss.item(),choose])
    min_emb.sort()
return min_emb[:10]

**Comment by YunShen66666** [comment]
this is my retrieval code
```
def find(img,path,model):
    target_img_emb = getEmbedding(img,'cpu',model)
    min_emb = []
    for i in range(100):
        choose = random.randint(0,30000)
        img_path = path + "\\" + f"{str(choose).zfill(5)}.jpg"
        img1 = cv2.imread(img_path)
        img1_emb = getEmbedding(img1,'cpu',model)
        loss = F.mse_loss(img1_emb,target_img_emb)
        min_emb.append([loss.item(),choose])
    min_emb.sort()
    return min_emb[:10]
```

**Comment by RiweiChen** [comment]
> this is my retrieval code
> 
> ```
> def find(img,path,model):
>     target_img_emb = getEmbedding(img,'cpu',model)
>     min_emb = []
>     for i in range(100):
>         choose = random.randint(0,30000)
>         img_path = path + "\\" + f"{str(choose).zfill(5)}.jpg"
>         img1 = cv2.imread(img_path)
>         img1_emb = getEmbedding(img1,'cpu',model)
>         loss = F.mse_loss(img1_emb,target_img_emb)
>         min_emb.append([loss.item(),choose])
>     min_emb.sort()
>     return min_emb[:10]
> ```

I think your should norm the embedding feature?

**Comment by AmirSh15** [comment]
Hi,
The code seems alright by me.

**Comment by YunShen66666** [comment]
> > this is my retrieval code
> > ```
> > def find(img,path,model):
> >     target_img_emb = getEmbedding(img,'cpu',model)
> >     min_emb = []
> >     for i in range(100):
> >         choose = random.randint(0,30000)
> >         img_path = path + "\\" + f"{str(choose).zfill(5)}.jpg"
> >         img1 = cv2.imread(img_path)
> >         img1_emb = getEmbedding(img1,'cpu',model)
> >         loss = F.mse_loss(img1_emb,target_img_emb)
> >         min_emb.append([loss.item(),choose])
> >     min_emb.sort()
> >     return min_emb[:10]
> > ```
> 
> I think your should norm the embedding feature?

Did _norm the embedding feature_ mean that I should add a **batchnormlize** layer before FECNet output?

**Comment by AmirSh15** [comment]
It's better to divide them to their norm.
